src/test/java/pages/GoogleSearchResultsPage.java

```diff
@@ -0,0 +1,42 @@
+package pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindAll;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+public class GoogleSearchResultsPage extends AbstractPage {
+
+    @FindAll({@FindBy(xpath = "//cite[contains(text(), 'wikipedia')]")})
+    private List<WebElement> generalSearchResults;
+
+    @FindBy(xpath = "//div[@id='rso']/div[2]//cite")
+    private WebElement firstEnglishWikipediaLink;
+
+    protected AbstractPage openPage() {
+        return null;
+    }
+
+    public GoogleSearchResultsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public String contentOfFirstEnglishWikipediaLink() {
+        waitForPageLoad(driver);
+        return firstEnglishWikipediaLink.getText();
+    }
+
+    public int countSearchResults() {
+        waitForPageLoad(driver);
+        return generalSearchResults.size();
```

Comment:
comment above about return statement

Reply:
Refactored, thank you.
Please see in the following commit.